src/components/table.js

```diff
@@ -56,117 +54,100 @@ function Table(props) {
         });
   };
   // make a axios call every 10minutes
-  setInterval(getDistricts, 600000);
-
-  const doSort = (e, props) => {
-    const totalRow = states.splice(0, 1);
-    {/* console.log(totalRow);*/}
-    states.sort((StateData1, StateData2) => {
-      const sortColumn = sortData.sortColumn;
-      let value1 = StateData1[sortColumn];
-      let value2 = StateData2[sortColumn];
-
-      if (sortColumn != 'state') {
-        value1 = parseInt(StateData1[sortColumn]);
-        value2 = parseInt(StateData2[sortColumn]);
-      }
-
-      if (sortData.isAscending) {
-        return value1 > value2 ? 1 : (value1 == value2) && StateData1['state'] > StateData2['state'] ? 1 : -1;
-      } else {
-        return value1 < value2 ? 1 : (value1 == value2) && StateData1['state'] > StateData2['state'] ? 1 : -1;
-      }
+  // setInterval(getDistricts, 600000);
+  useEffect(() => {
+    const interval = setInterval(getDistricts, 600000);
+    return () => clearInterval(interval);
+  }, []);
+
+  const sortByKey = (arr, key, isAscending) => {
+    return arr.concat().sort((s1, s2) => {
+      const v1 = key !== 'state' ? parseInt(s1[key]) : s1[key];
+      const v2 = key !== 'state' ? parseInt(s2[key]) : s2[key];
+      return (v1 > v2 ? 1 : (v1 == v2) && s1['state'] > s2['state'] ? 1 : -1) * (isAscending ? 1 : -1);
     });
-    {/* console.log(states);*/}
-    states.unshift(totalRow[0]);
   };
 
-  const handleSort = (e, props) => {
-    const currentsortColumn = e.currentTarget.querySelector('abbr').getAttribute('title').toLowerCase();
-    const isAscending = sortData.sortColumn == currentsortColumn? !sortData.isAscending : sortData.sortColumn === 'state';
+  useEffect(() => {
+    const data = props.states.slice(1, props.states.length);
```

Comment:
You don't need to mention props.state.length. It automatically slices till the end if you omit it. 

```
props.states.slice(1)
```
This is good enough

Reply:
Done!!